02_bubble/brnbke/main_test.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"bytes"
+	"reflect"
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote("[1 2 3 5]\n")
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main(), expected %v, sadly got %v", expected, actual)
+	}
+}
+
+var testCases = map[string]struct {
+	input []int
+	want  []int
+}{
+	"empty":           {input: []int{}, want: []int{}},
+	"single value":    {input: []int{0}, want: []int{0}},
+	"positive values": {input: []int{1, 6, 2, 0}, want: []int{0, 1, 2, 6}},
+	"negative values": {input: []int{-1, -6, -2, 0}, want: []int{-6, -2, -1, 0}},
+	"already sorted":  {input: []int{0, 1, 2, 6}, want: []int{0, 1, 2, 6}},
+	"reverse sorted":  {input: []int{6, 2, 1, 0}, want: []int{0, 1, 2, 6}},
+}
+
+func TestBubble(t *testing.T) {
+	for name, test := range testCases {
+		test := test
+		var buf bytes.Buffer
+
+		out = &buf
+		t.Run(name, func(t *testing.T) {
+			b := bubble(test.input)
+			result := reflect.DeepEqual(test.want, b)
+			if !result {
+				t.Errorf("expected %v, got %v", test.want, b)
+			}
+		})
+	}
+}
+
+func TestInsertion(t *testing.T) {
+	for name, test := range testCases {
+		test := test
+		var buf bytes.Buffer
+		out = &buf
+		t.Run(name, func(t *testing.T) {
+			b := insertion(test.input)
+			result := reflect.DeepEqual(test.want, b)
+			if !result {
+				t.Errorf("expected %v, got %v", test.want, b)
+			}
+		})
+	}
+}
+
+func TestMergesort(t *testing.T) {
+	for name, test := range testCases {
+		test := test
+		var buf bytes.Buffer
+		out = &buf
+		t.Run(name, func(t *testing.T) {
+			b := mergesort(test.input)
+			result := reflect.DeepEqual(test.want, b)
+			if !result {
+				t.Errorf("expected %v, got %v", test.want, b)
+			}
+		})
+	}
+}
+
+func TestMerge(t *testing.T) {
+	var testCases = map[string]struct {
+		left  []int
```

Comment:
Similar to the above comment, this could be simplified to `left, right, want []int` as a single line